Fixed the dont_merge_cookies example in docs (#6077)
scrapy/scrapy: docs/topics/request-response.rst

        cookies for that domain and will be sent again in future requests.
        That's the typical behaviour of any regular web browser.

>         It is recommended not to set the ``dont_merge_cookies`` key to ``True``
>         in :attr:`request.meta <scrapy.Request.meta>` as it causes custom cookies to be ignored.

Reviewer: I would change the wording, as I think it could be understood as a general recommendation rather than something that only applies when custom cookies are set (even though it can be inferred from context), and it would be good to link `dont_merge_cookies` to its docs (even though the previous text did not).

What about this?

```suggestion
        Note that setting the :reqmeta:`dont_merge_cookies` key to ``True`` in
        :attr:`request.meta <scrapy.Request.meta>` causes custom cookies to be
        ignored.
```

Author: Yeah I'll change it and also add the link

Author: @Gallaecio is this the right [link](https://doc.scrapy.org/en/latest/topics/downloader-middleware.html#cookies-enabled) for the `dont_merge_cookies` docs

Reviewer: `:reqmeta:`dont_merge_cookies` will generate a link, it’s a special syntax we have in Scrapy docs for request metadata keys.

        cookies for that domain and will be sent again in future requests.
        That's the typical behaviour of any regular web browser.

        Example of a request that sends manually-defined cookies:

        .. code-block:: python

            Request(
                url="http://www.example.com",
                cookies={"currency": "USD", "country": "UY"},

>             )

Reviewer: The only purpose of this example here was to feature dont_merge_cookies, since there is already an example of cookie use higher up. So I think you can remote this example (and the introductory line) altogether.

Author: Sure, I'll remove it.
